### PR TITLE
Bach[Athena]: use ordinal position for athena grouby clause

### PR DESCRIPTION
**Pull request by KathiaBarahona**
This fixes issue of grouping by with complex expressions

### CODE REVIEW

File: bach/bach/partitioning.py [hunk 1]
```diff
@@ -157,6 +157,16 @@ def get_index_expressions(self) -> List[Expression]:
             # this logic does not applies in Postgres
             if is_bigquery(idx.engine):
                 exprs.append(Expression.column_reference(idx.name))
+
+            # Athena allows grouping by expressions, column names and ordinals. A problem arises
+            # when the series' expression is way too complex to handle (expression dependency)
+            # In order to avoid this type of situations, is best to use ordinals, as we can make reference
+            # to complex expressions by just using there ordinal position in the SELECT clause.
+            # By default, index columns are the first in the SELECT clause. So, we can trust the order of
+            # the Grouby.index.
```
Comment (jansentom) [inline bach/bach/partitioning.py:166]:
```suggestion
            # the GroupBy.index.
```

File: bach/bach/partitioning.py [hunk 2]
```diff
@@ -142,7 +142,7 @@ def __eq__(self, other):
     def get_index_expressions(self) -> List[Expression]:
         from bach.series import SeriesAbstractMultiLevel
         exprs = []
-        for idx in self.index.values():
+        for pos, idx in enumerate(self.index.values()):
             if isinstance(idx, SeriesAbstractMultiLevel):
```
Comment (jansentom) [inline bach/bach/partitioning.py:146]: whoa, good one.

File: bach/bach/partitioning.py [hunk 3]
```diff
@@ -157,6 +157,16 @@ def get_index_expressions(self) -> List[Expression]:
             # this logic does not applies in Postgres
```
Comment (jansentom) [inline bach/bach/partitioning.py:157]: this comment is better placed in the `else`?

**Review by jansentom — APPROVED** [review]
This is really nice! Some small comments.

File: bach/bach/partitioning.py (hunk 1 above)
Comment (KathiaBarahona) [reply bach/bach/partitioning.py]: fixed here https://github.com/objectiv/objectiv-analytics/pull/1359/commits/520359cf46b2b69c22f575acc62f96a01284761b

File: bach/bach/partitioning.py (hunk 3 above)
Comment (KathiaBarahona) [reply bach/bach/partitioning.py]: yes, https://github.com/objectiv/objectiv-analytics/pull/1359/commits/520359cf46b2b69c22f575acc62f96a01284761b

**Pull request merged**